modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureStore.java

```diff
@@ -70,6 +71,9 @@ public ShapefileFeatureStore(ContentEntry entry, ShpFiles files) {
                     new IndexedShapefileFeatureWriter(
                             ds.indexManager, reader, ds.getCharset(), ds.getTimeZone());
         } else {
+            if (ds.indexManager.isIndexStale(ShpFileType.FIX)) {
```

Comment:
This would create the fix file if missing, but if it's missing we have no fid remapping, so why waste time to create a fid index in the first place? At the very least, I'd check if the file exists (maybe isIndexStyle needs a flag to check if the file exists before stating it's outdate).